networkx/algorithms/isomorphism/tree_isomorphism.py

```diff
@@ -1,29 +1,1021 @@
+
+        # Move to the next level.
+        current_level += 1
+
+    # If all the previous levels are equal, then build an isomorphism with the
+    # previous information.
+    isomorphism = build_isomorphism(root_T1, root_T2, children_T1, children_T2)
+
+    return True, isomorphism
+
+
+def get_height(T, root):
+    """Returns the height of a rooted tree.
+
+    Parameters
+    ----------
+    T : NetworkX Graph
+        A rooted tree.
+
+    root : node
+        The root of the tree.
+
+    Returns
+    -------
+    height : int
+        The height of the rooted tree.
+
+    Note
+    ----
+    This algorithm runs in O(n) time and space.
+
+    """
+    # Define a map for the distance to the root.
+    distance_to_root = {}
+
+    # The max distance between a root and another vertex.
+    max_distance = 0
+
+    distance_to_root[root] = 0
+
+    # Perform a BFS traversal to determine the distances.
+    for (parent, child) in nx.bfs_edges(T, root):
+        distance_to_root[child] = distance_to_root[parent] + 1
+
+        if max_distance < distance_to_root[child]:
+            max_distance = distance_to_root[child]
+
+    return max_distance
+
+
+def get_parenthood(T, root):
+    """Returns the parenthood map of a rooted tree.
+
+    Parameters
+    ----------
+    T : NetworkX Graph
+        A rooted tree.
+
+    root : node
+        The root of the tree.
+
+    Returns
+    -------
+    parenthood : dict of node: node
+        The parenthood map of the rooted tree.
+
+    Note
+    ----
+    This algorithm runs in O(n) time and space.
+
+    """
+    # Define an empty parenthood map.
+    parenthood = {}
+
+    # Perform a BFS traversal to determine the parenthood map.
+    for (parent, child) in nx.bfs_edges(T, root):
+        parenthood[child] = parent
+
+    return parenthood
+
+
+def rooted_tree_isomorphism_n(T1, root_T1, T2, root_T2):
+    """Returns an isomorphism between two rooted trees, if it exists.
+
+    Parameters
+    ----------
+    T1 : NetworkX Graph
+        A rooted tree.
+
+    root_T1 : node
+        The root of the rooted tree T1.
+
+    T2 : NetworkX Graph
+        A rooted tree.
+
+    root_t2 : node
+        The root of the rooted tree T2.
+
+    Returns
+    -------
+    are_iso : bool
+        Returns True if the given rooted trees are isomorphic, False otherwise.
+
+    isomorphism : dict of node of T1: node of T2
+        The isomorphism mapping f such that (u, v) is in E_T1 if and only if
+        (f(u), f(v)) is in E_T2. This dictionary is empty when there is no
+        isomorphism between the trees.
+
+    Note
+    ----
+    This algorithm runs in O(n) time and space.
+
+    """
+    # If both trees have different amount of vertices, return false.
+    if T1.order() != T2.order():
+        return False, {}
+
+    # If both trees are empty, return true.
+    if T1.order() == 0:
+        return True, {}
+
+    # If trees have more than one node, verify that they're indeed trees.
+    assert nx.is_tree(T1)
```

Comment:
I know it was already the case in the previous code, but a more explicit error message could be useful here.